File: SingleNumber.py
# single number

class Solution:
    # @param A, a list of integer
    # @return an integer
    def singleNumber(self, A):
        x = A[0]
        for i in range(1,len(A)):
            x = x^A[i]

        return x



if __name__ == "__main__":
    s=Solution()
    A=[1,1,2,3,4,5,4,3,5]
    r=s.singleNumber(A)
    print(r)

File: SingleNumber.cpp
/*

Given an array of integers, every element appears twice except for one. Find that single one.

Note:
Your algorithm should have a linear runtime complexity. Could you implement it without using extra memory?

*/

#include <iostream>

using namespace std;

class Solution {
public:
    int singleNumber(int A[], int n) {
        int x=A[0];
        for(int i=1;i<n;i++)        
            x ^=A[i];//异或可以将出现次数为偶数的清零
        return x;
    }
};

void main(){

Solution so;

A[7]={1,1,2,4,3,4,3};n=7;

x= so.singleNumber(A,n);

cout<<x<<endl;

}
